pkg/test/protocol_test_service/proto/protocol_test_service.proto

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package proto;
+
+// A simple service used for testing various protocols.
+service ProtocolTestService {
+    // Exchanges a request/response with the service.
+    rpc Exchange(ExchangeRequest) returns (ExchangeResponse);
+
+    // Sends a control request to the service, instructing it to perform a number of exchanges with another service.
+    rpc Control(ControlRequest) returns (ControlResponse);
+}
+
+// A header used by protocols that support headers (e.g. HTTP, GRPC).
+message Header {
+    string key = 1;
+    string value = 2;
+}
+
+enum Protocol {
+    HTTP = 0;
+    HTTP2 = 1;
+    GRPC = 2;
+    WEBSOCKET = 3;
+    TCP = 4;
+}
+
+message Transport {
+    Protocol protocol = 1;
+    bool secure = 2;
+}
```

Comment:
How is this expected to work given that we have HTTP1.1, WEBSOCKET, TCP, etc., and you are defining a full gRPC protocol for this?